### PR TITLE
312 update GitHub client

### PR DESCRIPTION
**Pull request by mbao01**
_Merge Order - 2_

### CODE REVIEW

**Comment by mbao01** [comment]
@yegor256 please review

File: 0pdd.rb [hunk 1]
```diff
@@ -213,11 +213,24 @@
...
+      name: name,
+      id_rsa: settings.config['id_rsa'],
+      dir: settings.temp_dir,
+      master: master || 'master'
+    )
+    repo.push
+    xml = repo.xml
+    xml.xpath('//processing-instruction("xml-stylesheet")').remove
+    xml.to_s
+  rescue StandardError
```
Comment (yegor256) [inline 0pdd.rb:231]: @mbao01 what is the point of this? why standard error reporting is not good enough?

File: objects/clients/github.rb [hunk 2]
```diff
@@ -49,5 +45,7 @@ def client
       puts "#{tp.defined_class}##{tp.method_id}()" if tp.defined_class == client.class
     end.enable
     client
+  rescue Octokit::NotFound
+    puts 'Issue with account not found'
```
Comment (yegor256) [inline objects/clients/github.rb:49]: @mbao01 oops...

Comment (mbao01) [reply objects/clients/github.rb:49]: Thank you, looking into it now

**Comment by mbao01** [comment]
@yegor256 please review

File: objects/tickets/github_tickets.rb [hunk 3]
```diff
@@ -40,7 +40,7 @@ def notify(issue, message)
       "@#{@github.issue(@repo, issue)['user']['login']} #{message}"
     )
   rescue Octokit::NotFound => e
-    puts "The issue most probably is not found, can't coment: #{e.message}"
+    puts "The issue most probably is not found, can't comment: #{e.message}"
```
Comment (mbao01) [inline objects/tickets/github_tickets.rb:43]: I only updated `coment` to `comment` on this line

**Comment by mbao01** [comment]
@yegor256 please review

File: objects/clients/github.rb [hunk 4]
```diff
@@ -24,18 +24,14 @@
...
 #
-class Github
-  def initialize(config = {})
-    @config = config
-  end
-
-  def client
-    client = if @config['testing']
+class GithubClient
+  def self.new(config = {})
```
Comment (yegor256) [inline objects/clients/github.rb:28]: @mbao01 what's wrong with a constructor? read this: https://www.yegor256.com/2017/11/14/static-factory-methods.html

Comment (mbao01) [reply objects/clients/github.rb:28]: Yes, I read through. Thank you. I will update it

**Comment by mbao01** [comment]
@yegor256 please review

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**